cvat/apps/dataset_manager/bindings.py

```diff
@@ -595,101 +939,189 @@ def _load_categories(cvat_anno, dimension):
+
+        dm_items: List[datumaro.DatasetItem] = []
+
+        ext_per_task: Dict[int, str] = {}
+        image_maker_per_task: Dict[int, Callable] = {}
+
+        for task in project_data.tasks:
+            is_video = task.mode == 'interpolation'
+            ext_per_task[task.id] = FrameProvider.VIDEO_FRAME_EXT if is_video else ''
+            if include_images:
+                frame_provider = FrameProvider(task.data)
+                if is_video:
+                    # optimization for videos: use numpy arrays instead of bytes
+                    # some formats or transforms can require image data
+                    def _make_image(i, **kwargs):
+                        loader = lambda _: frame_provider.get_frame(i,
+                            quality=frame_provider.Quality.ORIGINAL,
+                            out_type=frame_provider.Type.NUMPY_ARRAY)[0]
+                        return Image(loader=loader, **kwargs)
+                else:
+                    # for images use encoded data to avoid recoding
+                    def _make_image(i, **kwargs):
+                        loader = lambda _: frame_provider.get_frame(i,
+                            quality=frame_provider.Quality.ORIGINAL,
+                            out_type=frame_provider.Type.BUFFER)[0].getvalue()
+                        return ByteImage(data=loader, **kwargs)
+                image_maker_per_task[task.id] = _make_image
+
+        for frame_data in project_data.group_by_frame(include_empty=True):
+            image_args = {
+                'path': frame_data.name + ext_per_task[frame_data.task_id],
+                'size': (frame_data.height, frame_data.width),
+            }
+            if include_images:
+                dm_image = image_maker_per_task[frame_data.task_id](frame_data.idx, **image_args)
+            else:
+                dm_image = Image(**image_args)
+            dm_anno = self._read_cvat_anno(frame_data, project_data.meta['project']['labels'])
+            dm_item = datumaro.DatasetItem(id=osp.splitext(frame_data.name)[0],
+                annotations=dm_anno, image=dm_image,
+                subset=frame_data.subset,
+                attributes={'frame': frame_data.frame}
+            )
+            dm_items.append(dm_item)
+
+        self._items = dm_items
+
 
-            return dm_attr
+def GetCVATDataExtractor(instance_data: Union[ProjectData, TaskData], include_images: bool=False):
+    if isinstance(instance_data, ProjectData):
+        return CVATProjectDataExtractor(instance_data, include_images)
+    else:
+        return CvatTaskDataExtractor(instance_data, include_images)
+
+class CvatImportError(Exception):
+    pass
+
+def mung_image_name(name: str, images: List[Literal["path", "width", "height", "subset"]]) -> str:
+    name, ext = name.split(osp.extsep)
+    if not any(map(lambda image: image["path"] == name, images)):
+        return name + osp.extsep + ext
```

Comment:
Maybe you'd want to use
```suggestion
        return osp.extsep.join(parts)
```